src/olympia/abuse/models.py

```diff
@@ -354,12 +354,21 @@ def appeal(self, *, abuse_report, appeal_text, user, is_reporter):
             if not user:
                 appealer_entity = self.get_cinder_reporter(abuse_report)
         else:
+            # If the appealer is not an original reporter, we have to provide
+            # an authenticated user that is the author of the content.
+            # User bans are a special case, since the user can't log in any
+            # more at the time of the appeal, so we let that through using the
+            # target of the job that banned them.
+            if (
+                not user
+                and self.decision_action == CinderJob.DECISION_ACTIONS.AMO_BAN_USER
+            ):
+                user = self.target
```

Comment:
should we verify that self.target is actually a UserProfile instance at this point, or are we trusting that the data won't ever be that bad

Reply:
Let's be a little more robust just in case b38d7b00f7d60fd2cee009cb916bf407980f4c4e